packages/component-library/src/components/plmg-slider/plmg-slider.tsx

```diff
@@ -137,84 +154,106 @@ export class Slider {
   @State() inputFieldValue: number;
   @State() value: number;
   @State() internalRangeValues: number[];
+  @State() allowedInputs: number[];
 
   /**
    * The event "valueUpdated" is triggered when the slider value changes either by moving the thumb or entering in the text field.
    */
   @Event() valueUpdated: EventEmitter;
+  // /**
+  //  * The event "valueControlFailure" is emitted by when value passed
+  //  */
+  // @Event({ eventName: 'valueControlReject' })
+  // valueControlReject: EventEmitter;
 
   private stringToNumberArray(newValue: string) {
     return newValue.split(',').map(Number);
   }
 
-  private handleSliderChange(ev) {
-    this.updateValue(ev.target.value);
+  private setAllowedInputs() {
+    const range = Array.from(
+      { length: (this.max - this.min) / this.stepValue + 1 },
+      (_, i) => this.min + i * this.stepValue
+    );
+    return range.map(
+      (value) => Math.round((value + Number.EPSILON) * 1000000) / 1000000
+    );
```

Comment:
You have some fancy logic going on here. Could you explain what it is doing?

Reply:
I would very much like to see your explanation 😃 I want to be sure that I (and you) completely understand what the code is doing and how.

Reply:
Decided to write a [blog post](https://dev.to/jamesducky/explained-range-generator-with-arrayfrom-3eh2-temp-slug-4862945?preview=c72a5cd08858c0b5aeeb6f3a4c0347e38c94481406cb53519aa318f1db7e02ad705f3372b27ab3632ecdf4c842ebd7509598a77b2639c291e18e1b4d) on the subject 😃